playbooks/server-install.yml

```diff
@@ -135,3 +136,64 @@
         key: "{{ keys }}"
         state: present
         exclusive: true
+
+    - name: uninstall docker if it exists
+      shell: |
+        for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do apt-get remove $pkg; done
+
+    - name: Create keyrings directory
+      file:
+        path: /etc/apt/keyrings
+        state: directory
+        mode: 0755
+
+    - name: Add Docker GPG apt Key
+      get_url:
+        url: https://download.docker.com/linux/debian/gpg
+        dest: /etc/apt/keyrings/docker.asc
+        mode: 0644
+
+    - name: Add Docker repository
+      apt_repository:
+        repo: "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian {{ ansible_lsb.codename }} stable"
+        filename: docker
+        state: present
+
+    - name: Install docker
+      apt:
+        name:
+          - docker-ce
+          - docker-ce-cli
+          - containerd.io
+          - docker-buildx-plugin
+          - docker-compose-plugin
+        state: present
+        update_cache: yes
+
+    - name: Ensure /opt/nginx directory exists
+      file:
+        path: /opt/nginx
+        state: directory
+        owner: root
+        group: root
+        mode: 0755
+
+    - name: Create docker compose file
+      template:
+        src: ../templates/docker-compose.j2
+        dest: /opt/nginx/docker-compose.yaml
+        owner: root
+        group: root
+        mode: 0644
+
+    - name: Create nginx cache file
+      template:
+        src: ../templates/cache.conf.j2
+        dest: /opt/nginx/cache.conf
+        owner: root
+        group: root
+        mode: 0644
+
+    - name: Start nginx docker image
```

Comment:
This will _mostly_ work. I've usually used a systemd unit to keep things running: https://askubuntu.com/a/1459176

Reply:
I think we should switch it over. Let me know if you have any issues during the switch. Ideally we can stop/start via systemd too. This way monitoring tools like node exporter (prometheus based) can help us monitor the service too.